astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2082,6 +2097,70 @@ def test_unstructured_to_structured(self):
         # For the other tests of ``structured_to_unstructured``, see
         # ``test_structured.TestStructuredQuantityFunctions.test_unstructured_to_structured``
 
+    def test_merge_arrays(self):
```

Comment:
Parametrize this over `flatten` and have a different test only for the case it matters?

Reply:
☑ SGTM!